### ISSUE TITLE
CA****: Warn when using __ as separator in IWebHostBuilder.UseSetting or when setting keys in ConfigurationManager

### ISSUE THREAD

**Issue by modestotech** [dotnet/roslyn-analyzers#7827]
# Summary
`IWebHostBuilder.UseSetting` and `ConfigurationManager` uses `:` as the hierarchy separator for configuration keys. Using `__` (double underscore), which is the environment variable convention, is silently accepted but does not resolve to the expected configuration path, leading to configuration values not being bound correctly.

# Problem
Developers familiar with environment variable configuration may mistakenly use `__` in these cases:

```csharp
// WebApplicationBuilder
builder.UseSetting("Foo__Bar__0__Baz", "foobarbaz"); // ❌ silently wrong

// ApplicationBuilder
builder.Configuration["Foo__Bar__0__Baz"] = "foobarbaz"; // ❌ silently wrong
```

The correct separator is : `:`

```csharp
// WebApplicationBuilder
builder.UseSetting("Foo:Bar:0:Baz", "foobarbaz"); //  ✅

// ApplicationBuilder
builder.Configuration["Foo:Bar:0:Baz"] = "foobarbaz"; // ✅
```

There is no warning or error — the value is simply never bound, leading to null config objects or ValidateOnStart failures that are not always obvious when diagnosing.

# Proposed diagnostic
Use `:` instead of `__` as the hierarchy separator in UseSetting respectively ConfigurationManager. '__' is only supported for environment variables.

**Comment by Frulfump** [comment]
This seems useful to get some maintainer input on but have a look at the README in the repo

>Repository Consolidation
>
>The Microsoft.CodeAnalysis.NetAnalyzers package moved into the [dotnet/sdk](https://github.com/dotnet/sdk) repository for >further development and respond to issues formerly in this repository.
>
>All other analyzers moved into the [dotnet/roslyn](https://github.com/dotnet/roslyn) repository.

So it would be better to move this issue to the correct repo, please leave a link here :)
